apluslms_roman/builder.py

```diff
@@ -10,18 +10,24 @@
 
 class Builder:
     def __init__(self, engine, config, observer=None):
-        assert isdir(config.__path__), "Course configuration path doesn't exists"
+        if not isdir(config.dir):
+            raise ValueError("config.dir isn't a directory.")
         self.config = config
-        self.path = config.__path__
+        self.path = config.dir
         self._engine = engine
         self._observer = observer or StreamObserver()
 
-    def build(self):
+    def buildSteps(self):
+        return [BuildStep.from_config(i, step) for i, step in enumerate(self.config.steps)]
+
+    def build(self, step_list=[]):
```

Comment:
`, step_list=None` is bit more nice. If you like to tell the type, then we can write `, step_list: list = None`. Name `step_refs` might be a bit better.